test/system/260-sdnotify.bats

```diff
@@ -184,6 +184,65 @@ READY=1"
     _stop_socat
 }
 
+# These tests can fail in dev. environment because of SELinux.
+# quick fix: chcon -t container_runtime_exec_t ./bin/podman
+@test "sdnotify : healthy" {
```

Comment:
This test checks the behavior when using `health-cmd` which is more a health check than a readiness probe. Consider adding a test for `--health-startup-cmd` which is more suitable to what a user would try to achieve

Reply:
Good point.  It seems that `--health-startup-cmd` needs much more testing - it's not tested in `test/system` at all.

I will play a bit with your suggestion but think we need to improve the test coverage even further.

Reply:
@ygalblum, can you elaborate on the exact requirements?  IIUIC, the startup "gates" the ordinary healthcheck.  So once the startup HC is good, the ordinary HC is executed.  Only the latter influences the health status.

How do you want it to behave?

Reply:
I think I also share your understanding about the order of things in the code. As their name suggests, the startup checks verify that the service has started correctly. Once it has, the health checks kick in to check that the service is in fact still running.

As for my comment, if I look at it from systemd's perspective, it need to know that the service has started which IMHO should correspond with the startup checks more than with the health check.

Having said that, I guess we would like to allow users to define only the health check and let it act as the startup one. But, regardless to whether the health check is set, if the startup health is set, I think that systemd should be notified once it is successful.

Again, from the looks of the code, I think it does that, but I'm not sure. So, we should test it. In addition, as always, tests are there not only to make sure that the code we added it correct, but also to make sure the behavior is not broken unintentionally.

Reply:
> tests are there not only to make sure that the code we added it correct, but also to make sure the behavior is not broken unintentionally.

Totally agree.  That's why I am holding the PR.

> As for my comment, if I look at it from systemd's perspective, it need to know that the service has started which IMHO should correspond with the startup checks more than with the health check.

That is currently not possible. The `wait --condition=healthy` logic waits for the main HC, not the startup one.  The latter has no impact on the health status other than it will delay the main HC from running.

So in other words: you are looking for a means to wait for the startup HC to finish?  If that's the case, then I think we can merge this PR as is but also need to take a step back and see how we can cover the startup one.

Reply:
Something like `wait --condition=$startup_done` with a nicer name for `$startup_name`.  Once that's done, we can add a `--sdnotify=$startup_done` on top?

Reply:
BTW. I really like the idea as it - as you mentions - discriminates the initialization/startup phase from the workload phase.

Reply:
The way I read this feature request is to have podman wait until the service inside the container has finished its startup sequence before notifying systemd that it is ready. As such, I would assume that the startup health checks are the ones driving it.
And yes, the idea is to separate between "succeeded/failed to start" and runtime failures. 
Having said that, if waiting for only the startup is not supported, this PR can be merged

Reply:
Thanks for confirming, @ygalblum!  Let's chat about how to support startup HCs in tomorrow's meeting.